### PR TITLE
Hotfix: AWS S3 object store needs explicit credentials

### PR DESCRIPTION
**Pull request by zaychenko-sergei**

### CODE REVIEW

File: kamu-core/Cargo.toml [hunk 1]
```diff
@@ -56,6 +56,7 @@ sha3 = "0.10"
 aws-config = { version = "0.55" }
 aws-sdk-s3 = { version = "0.28" }
 aws-smithy-http = { version = "0.55", features = ["rt-tokio"] }
+aws-credential-types = "*"
```
Comment (sergiimk) [inline kamu-core/Cargo.toml:59]: Please specify a version - I'm making `cargo deny` rules more strict to reject the wildcard dependencies.

File: kamu-core/tests/tests/test_query_service_impl.rs [hunk 2]
```diff
@@ -138,8 +140,8 @@ async fn create_catalog_with_s3_workspace(s3: &S3) -> dill::Catalog {
         .add_value(infra::ObjectStoreBuilderLocalFs::new())
         .bind::<dyn ObjectStoreBuilder, infra::ObjectStoreBuilderLocalFs>()
         .add_value(infra::ObjectStoreBuilderS3::new(
-            s3_context.bucket,
-            endpoint.unwrap(),
+            s3_context,
+            s3_credentials,
```
Comment (sergiimk) [inline kamu-core/tests/tests/test_query_service_impl.rs:144]: Do we need to pass credentials separately if they can already be obtained from `S3Context`?

**Review by sergiimk — APPROVED** [review]

File: kamu-core/Cargo.toml (hunk 1 above)
Comment (zaychenko-sergei) [reply kamu-core/Cargo.toml:59]: Done

File: kamu-core/tests/tests/test_query_service_impl.rs (hunk 2 above)
Comment (zaychenko-sergei) [reply kamu-core/tests/tests/test_query_service_impl.rs:144]: Yes, because it's async operation, and the context around is synchronous

**Pull request merged**